lib/engine.py

```diff
@@ -541,11 +540,12 @@ def format_insert_value(self, value, datatype):
         elif datatype == "char":
             if strvalue.lower() in nulls:
                 return "null"
-            # automatically escape quotes in string fields
-            if hasattr(self.table, "escape_double_quotes") and self.table.escape_double_quotes:
-                strvalue = self.escape_double_quotes(strvalue)
-            if hasattr(self.table, "escape_single_quotes") and self.table.escape_single_quotes:
-                strvalue = self.escape_single_quotes(strvalue)
+            if escape:
+                # automatically escape quotes in string fields
+                if hasattr(self.table, "escape_double_quotes") and self.table.escape_double_quotes:
+                    strvalue = self.escape_double_quotes(strvalue)
+                if hasattr(self.table, "escape_single_quotes") and self.table.escape_single_quotes:
+                    strvalue = self.escape_single_quotes(strvalue)
```

Comment:
It's not clear to me from the code why we would want to escape sometimes and not others. Can you add some description of this to the docstring or in the inline comment on 544.